### PR TITLE
FP-58 added cartService with logic for add product to file and get ca…

### PR DESCRIPTION
**Pull request by natam**
FP-58 added cartService with logic for add product to file and get ca…

### CODE REVIEW

File: src/main/java/com/nkh/ECommerceShop/service/CartsService.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import com.nkh.ECommerceShop.security.service.UserDetailsImpl;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class CartsService {
```
Comment (moseseth) [inline src/main/java/com/nkh/ECommerceShop/service/CartsService.java:17]: adding unit test for services is highly recommended

File: src/main/java/com/nkh/ECommerceShop/service/CartsService.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        Product product = productsService.getById(productId);
+        if (productsService.checkProductStock(productId, quantity)) {
+            Cart myCart = getMyCart();
+            Optional<CartProduct> foundCartProduct = myCart
+                    .getCartProducts()
+                    .stream()
+                    .filter(cartProducts -> cartProducts.getProduct()
+                            .getId() == productId)
+                    .findFirst();
+            if (foundCartProduct.isPresent()) {
```
Comment (moseseth) [inline src/main/java/com/nkh/ECommerceShop/service/CartsService.java:43]: What does this section do?

Comment (natam) [reply src/main/java/com/nkh/ECommerceShop/service/CartsService.java:43]: check if product presents in cart -> increase it's quantity, if not add product to cart

File: src/main/java/com/nkh/ECommerceShop/service/CartsService.java (hunk 1 above)
Comment (natam) [reply src/main/java/com/nkh/ECommerceShop/service/CartsService.java:17]: I have separate ticket for tests for each userstory

**Review by faruka-startsteps — APPROVED** [review]

**Pull request merged**